src/process.rs

```diff
@@ -505,6 +526,52 @@ async fn process_cloudtrail(
     }
 }
 
+pub async fn kafka_logs(
+    topic: String,
+    records:  Vec<KafkaRecord>,
+    coralogix_exporter: DynLogExporter,
+    config: &Config,
+) -> Result<(), Error> {
+    let defined_app_name = config
+        .app_name
+        .clone()
+        .unwrap_or_else(|| "NO APPLICATION NAME".to_string());
+
+    let defined_sub_name = config
+        .sub_name
+        .clone()
+        .unwrap_or(topic);
+
+    let mut batch = Vec::new();
+    for record in records {
+        if record.value.is_none() {
+            continue
+        }
+
+        // check if value is base64 encoded
+        if let Ok(message) = BASE64_STANDARD.decode(record.value.clone().unwrap()) {
+            batch.push(String::from_utf8(message)?);
```

Comment:
Is it expected to fail if the decoded message is not utf-8?

Reply:
@rtimush , in some cases kafka could be sending a raw none-base64 encoded string, so the goal is to check if it can be successfully decoded as base64, if not, a raw string is assumed.

Reply:
What I mean is that if a value is base64-encoded but it's not a valid utf-8 string, the conversion will return an error. I was thinking that maybe using the raw string in this case is preferable.

Reply:
Thanks for clarifying @rtimush 

I believe all strings in rust must be utf-8 compatible. There's no way  (_that I know of_) to convert a `Vec<u8>` to anything but a utf-8 string. 

If it fails to decode a valid utf-8 string then it can be assumed we're dealing with bad data that wouldn't be compatible with what is required for our backend. So returning an `err` here seems like the best course. 

Otherwise, let me know if i've misunderstood.

Reply:
Basically, I'm asking about the `String::from_utf8(message) == Err(_)` case — should it be an error, or should we send `record.value` we had before base64 decoding. I don't know the answer, just making sure returning an error is intentional.

Reply:
If we get to the`String::from_utf8(message)` it means that we've successfully decoded Base64. So if it fails to convert the bytes to a utf8 string after, and we return the original value and ignore the error, it would mean we're passing a Base64 encoded string to our backend.